### DISCUSSION TITLE
General Gas fee of deploy simple smart contract

### DISCUSSION THREAD

**Discussion by DaoSpirit** [zkSync-Community-Hub/zksync-developers#212] [General] [answered]
### Environment

Mainnet

### Provide a brief description of the functionality you're trying to implement.

how much does it cost to deploy simple smart contract to zkSync mainnet?
I can find transaction hash on explorer and the gas fee is about $50 - 100, in general.
zkSync is layer2 and the gas fee will be very smaller than layer1.
plz share your opinion, if possible.
thanks

### What is the specific issue or error you're encountering?

Estimate gas of deployment

### Can you share the error messages or logs you're receiving, if any?

N/A

### Have you made any recent changes to the contract before encountering this issue?

N/A

### Are there any external libraries or contracts that your contract interacts with?

N/A

### Can you provide the relevant portions of your contract code where the issue is occurring?

N/A

### Have you tried to isolate the problem, and if so, what were the results?

N/A

### What steps have you already taken to try to resolve the issue?

N/A

**Comment by dutterbutter** [comment, marked as answer]
Much cheaper then you've outlined but it depends on the contract. I would suggest deploying a contract to see the costs :) For example a simplified oracle contract costs `~0.0001145415 eth` but it entirely depends on the contract.

**Reply by DaoSpirit** [reply]
thanks.
then what about deploying simple ERC20 contract?

**Comment by dougtende** [comment]
Deploying a simple smart contract to zkSync mainnet typically incurs significantly lower gas fees compared to Layer 1 Ethereum. While the exact cost can vary based on factors such as contract complexity and network congestion, you can generally expect the gas fees to be much smaller, often ranging from a few cents to a couple of dollars. This makes zkSync an attractive option for deploying smart contracts, especially for applications where minimizing transaction costs is important. However, it's always a good idea to use tools like gas estimators or testnets to get a more accurate estimate before deploying your contract to the mainnet.

**Comment by go4comdathern** [comment]
Based on typical gas costs on zkSync mainnet, deploying a simple smart contract can cost significantly less compared to Layer 1 Ethereum. Expect deployment fees ranging from a few cents to a couple of dollars, making it a cost-effective option for deploying contracts.

**Comment by lynnrosado** [comment]
Deploying a simple smart contract to zkSync mainnet typically costs much less than on layer1 Ethereum. Gas fees can vary but are generally significantly lower, making zkSync an efficient and cost-effective solution for deploying smart contracts.

**Comment by tcaniao** [comment]
Deploying a simple smart contract to zkSync mainnet typically costs significantly less than on layer 1 Ethereum due to zkSync's scalability solutions. Gas fees are generally much smaller, making it cost-effective for deploying contracts.

**Comment by crissshawn** [comment]
The user is inquiring about the cost of deploying a simple smart contract to zkSync mainnet. They have observed that the gas fee is generally around $50 - $100 on the explorer, but are interested in opinions on the matter. They are specifically looking for an estimate of the gas cost of deployment. No error messages or logs were shared, and no recent changes to the contract were made. There are no external libraries or contracts that the contract interacts with, and no steps have been taken to try to resolve the issue. No repository link was provided.

**Comment by eddiedenna** [comment]
The user is inquiring about the cost of deploying a simple smart contract to zkSync mainnet and estimates the gas fee to be around $50-$100. They are seeking opinions on the matter and do not provide any error messages or logs. The user has not made any recent changes to the contract and is not interacting with any external libraries or contracts. They have not tried to isolate the problem and have not taken any steps to resolve the issue. The user does not provide a repo link.
